kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/psi/PsiProvider.kt

```diff
@@ -38,11 +44,34 @@ object PsiProvider {
     }
 
     fun collectPsiMethods(psiFile: PsiFile): MutableCollection<KtNamedFunction> {
-        return PsiTreeUtil.collectElementsOfType(psiFile, KtNamedFunction::class.java)
+        return collectElementsOfType(psiFile, KtNamedFunction::class.java)
+    }
+
+    fun collectPsiImportDirectives(psiFile: PsiFile): MutableCollection<KtImportDirective> {
+        return collectElementsOfType(psiFile, KtImportDirective::class.java)
+    }
+
+    private fun <T : PsiElement> extractElementsOfTypeFromProject(
+        project: Project,
+        psiElementClass: Class<T>
+    ): List<T> {
+        // We should reverse the list with method since we handle all elements separately
+        // and we can invalidate the previous one
+        return extractPsiFiles(project)
+            .map { collectElementsOfType(it, psiElementClass) }
+            .flatten()
+            .reversed()
+    }
+
+    private fun <T : PsiElement> collectElementsOfType(
```

Comment:
I'm not sure if there is need for this function since you can just use PsiTreeUtil.collectElementsOfType()

Reply:
Remove it